### PR TITLE
Add xkcd command

### PR DESCRIPTION
**Pull request by TheDevFreak**
# Initial Checklist
- [x] Has @tigattack and/or @kdavis been added as a reviewer?
- [ ] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [x] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/develop/README.md#cog-summaries)?

# Details
* **Does this resolve an issue?**
Resolves 
image

### New Features
* Adds xkcd command which gets specified comic or latest comic if none is specified

### Breaking Changes
* N/A

### CODE REVIEW

**Comment by tigattack** [comment]
Add to the ToC in readme please.

File: xkcd/xkcd.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from redbot.core import commands
+import discord
+import random
+import asyncio,aiohttp,async_timeout,json
+
+
+#Define function to do web requests
+async def fetchGet(urlIn):
```
Comment (issy) [inline xkcd/xkcd.py:8]: Can you please make your vars and funcs snake_case to align with pep8? kthxbai

**Review by issy — CHANGES_REQUESTED** [review]

**Review by tigattack — APPROVED** [review]
Looks good to me but what do I know 🙂

File: xkcd/xkcd.py (hunk 1 above)
Comment (TheDevFreak) [reply xkcd/xkcd.py]: 4fc53da

File: xkcd/xkcd.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        #Load as json
+        comicJson = json.loads(comicJson)
+
+        #Build embed for xkcd comic
+        
+        xkcdEmbed = discord.Embed(title='xkcd Comic: #{}'.format(comicJson["num"]), colour=ctx.guild.me.colour)
+        xkcdEmbed.add_field(name="Comic Title", value=comicJson["safe_title"])
+        xkcdEmbed.add_field(name="Publish Date", value=f"{comicJson['year']}-{comicJson['month']}-{comicJson['day']}")
+        #If there is alt text add it to the embed, otherwise don't
+        if comicJson["alt"] != "":
```
Comment (issy) [inline xkcd/xkcd.py:47]:
```suggestion
        if not comicJson["alt"]:
```
This is more pythonic, as empty strings are falsy

**Review by issy — CHANGES_REQUESTED** [review]

File: xkcd/xkcd.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    def __init__(self):
+        pass
+
+    @commands.command()
+    async def xkcd(self, ctx, comicNumber: int = 0):
+        """Returns xkcd comic of given number, otherwise return latest comic."""
+        
+        #Because Randall is very funny comic #404 doen't exist
+        if comicNumber == 404:
```
[the remark this answers is not in the document]
Comment (TheDevFreak) [reply xkcd/xkcd.py]: 726d746

File: xkcd/xkcd.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        if comicNumber == 0:
+            #No comic specified get latest
+            url = "https://xkcd.com/info.0.json"
+        else:
+            url = f"https://xkcd.com/{comicNumber}/info.0.json"
+        
+        #Get comic data from xkcd api
+        comicJson = await fetch_get(url)
+        #Load as json
+        comicJson = json.loads(comicJson)
```
[the remark this answers is not in the document]
Comment (TheDevFreak) [reply xkcd/xkcd.py]: 726d746

File: xkcd/xkcd.py (hunk 2 above)
Comment (TheDevFreak) [reply xkcd/xkcd.py]: 4ff1d42

**Review by issy — APPROVED** [review]

**Comment by kdavis** [comment]
@issy 🤔 You good?

**Comment by tigattack** [comment]
> @issy 🤔 You good?

Not sure who she thinks she is tbh.

File: xkcd/xkcd.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        # If there is alt text add it to the embed, otherwise don't
+        if not data["alt"]:
+            xkcd_embed.add_field(name="Comic Alt Text", value=data["alt"])
+        else:
+            pass
+        xkcd_embed.set_image(url=data["img"])
+        return xkcd_embed
+
+    def make_error_embed(self, ctx: commands.Context, error_type: str) -> discord.Embed:
+        "Generate error message embeds"
```
Comment (kdavis) [inline xkcd/xkcd.py:58]: Can we use `"""` for docstring

File: xkcd/xkcd.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    def make_comic_embed(self, ctx: commands.Context, data: dict) -> discord.Embed:
+        """Generate embed for xkcd comic"""
+        xkcd_embed = discord.Embed(title=f"xkcd Comic: #{data['num']}", colour=ctx.guild.me.colour)
+        xkcd_embed.add_field(name="Comic Title", value=data["safe_title"])
+        xkcd_embed.add_field(name="Publish Date", value=f"{data['year']}-{data['month']}-{data['day']}")
+        # If there is alt text add it to the embed, otherwise don't
+        if not data["alt"]:
+            xkcd_embed.add_field(name="Comic Alt Text", value=data["alt"])
+        else:
```
Comment (kdavis) [inline xkcd/xkcd.py:52]: Do we need an `else: pass` here?

File: xkcd/xkcd.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            async with session.get(url_in) as response:
+                await session.close()
+                if response.status != 200:
+                    return {}
+                return await response.json()
+
+class Xkcd(commands.Cog):
+    """xkcd Cog"""
+
+    def __init__(self):
```
Comment (kdavis) [inline xkcd/xkcd.py:19]: We don't need init if you're not doing anything

**Pull request merged**